docs/reference_guide.md

```diff
@@ -2369,6 +2371,32 @@ ioctl(PERF_EVENT_IOC_SET_BPF): Invalid argument
 Error attaching probe: 'kprobe:vfs_read'
 ```
 
+## 21. `buf()`: Buffers
+
+Syntax: `buf(void *d, int length)`
+
+Returns a hex-formatted string of the data pointed to by `d` that is safe to print. Because the
+length of the buffer cannot be determined automatically in many cases, `length` must be provided to
+limit the number of bytes that are read. By default, the maximum number of bytes is 64, but this can
+be tuned using the [`BPFTRACE_STRLEN`](#7-env-bpftrace_strlen) environment variable.
+
+For example, we can take the `buff` parameter (`void *`) of `sys_enter_sendto`, read the
+number of bytes specified by `len` (`size_t`), and format the bytes in hexadecimal so that
+they don't corrupt the terminal display. The resulting string can be provided as an argument to
+printf() using the `%r` format specifier:
+
+```
+# bpftrace -e 'tracepoint:syscalls:sys_enter_sendto { printf("Datagram bytes: %r\n", buf(args->buff, args->len)); }' -c 'ping 8.8.8.8 -c1'
+Attaching 1 probe...
+PING 8.8.8.8 (8.8.8.8) 56(84) bytes of data.
+Datagram bytes: \x08\x00+\xb9\x06b\x00\x01Aen^\x00\x00\x00\x00KM\x0c\x00\x00\x00\x00\x00\x10\x11\x12\x13\x14\x15\x16\x17\x18\x19\x1a\x1b\x1c\x1d\x1e\x1f !"#$%&'()*+,-./01234567
+64 bytes from 8.8.8.8: icmp_seq=1 ttl=52 time=19.4 ms
```

Comment:
Please wrap the lines (100 char width? not sure) so that the example doesn't need a horizontal scroll bar